### ISSUE TITLE
Empty required components not getting highlighted

### ISSUE THREAD

**Issue by JanKoll** [strapi/strapi#18457]
### Required System information

- Node.js version: 20.8.1
- NPM version: 10.1.0
- Strapi version: 4.14.4
- Database: MySQL
- Operating system: MacOS
- Is your project Javascript or Typescript: TS

### Describe the bug

When you create a collection or single type, the highlighting for required Components does not work. This makes it difficult on larger pages for editors to create/save content.

### Steps to reproduce the behavior

Just create a collection or single type with a required component and try to save without initiating the component. You can't save the page and there is no hint provided.

### Expected behavior

A hint like on simple text fields.

**Comment by rahulbansod519** [comment]
Did you mean that if you click "create new collection type" without providing any data for the fields, there should be a visual indication, such as highlighting the HTML element, to show that a value is required?

**Comment by JanKoll** [comment]
Yes, on any other fields that work fine, but not for components.
![Screenshot 2023-10-16 at 14 50 54](https://github.com/strapi/strapi/assets/33149112/3d790465-4aff-4616-b2f5-4f8e04e95d28)
